File: LimFTPClient/OpenNetCF/Samples/VB/FTPTestVB/Form1.vb
Imports OpenNETCF.Net.Ftp
Imports System.IO

Delegate Sub StringDelegate(ByVal value As String)

Public Class Form1
    Private WithEvents m_ftp As FTP

    Private Sub connect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles connect.Click
        OnResponse("Connecting")

        m_ftp = New FTP(server.Text)
        m_ftp.BeginConnect(user.Text, password.Text)
    End Sub

    Private Sub m_ftp_Connected(ByVal source As FTP) Handles m_ftp.Connected
        ' when this happens we're ready to send command
        OnResponse("Connected.")
    End Sub

    Private Sub m_ftp_ResponseReceived(ByVal source As FTP, ByVal Response As FTPResponse) Handles m_ftp.ResponseReceived
        OnResponse(Response.Text)
    End Sub

    Private Sub OnResponse(ByVal response As String)
        If (Me.InvokeRequired) Then

            Me.Invoke(New StringDelegate(AddressOf OnResponse), New Object() {response})
            Return
        End If

        Dim item As ListViewItem = New ListViewItem(New String() {DateTime.Now.ToShortTimeString(), response})
        status.Items.Insert(0, item)
        status.Columns(1).Width = -1
    End Sub

    Private Sub getFileList_Click(ByVal sender As Object, ByVal e As EventArgs) Handles getFileList.Click
        Dim files As FTPFiles = m_ftp.EnumFiles()

        fileList.Items.Clear()

        For Each file As FTPFile In files
            fileList.Items.Add(New ListViewItem(New String() {file.Name, file.Size.ToString()}))
        Next

        tabs.SelectedIndex = 1
    End Sub

    Private Sub upload_Click(ByVal sender As Object, ByVal e As EventArgs) Handles upload.Click
        Dim stream As FileStream = File.OpenRead("\\My Documents\\My Pictures\\Waterfall.jpg")
        m_ftp.SendFile(stream, "waterfall.jpg")
        stream.Close()
    End Sub
End Class

File: LimFTPClient/OpenNetCF/Samples/CS/FTPTest/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using OpenNETCF.Net.Ftp;
using System.IO;

namespace FTPTest
{
    public partial class Form1 : Form
    {
        delegate void StringDelegate(string value);

        private FTP m_ftp;

        public Form1()
        {
            InitializeComponent();
        }

        private void connect_Click(object sender, EventArgs e)
        {
            OnResponse("Connecting");
            m_ftp = new FTP("limowski.xyz", 2121);
            m_ftp.ResponseReceived += new FTPResponseHandler(m_ftp_ResponseReceived);
            m_ftp.Connected += new FTPConnectedHandler(m_ftp_Connected);
            m_ftp.BeginConnect(user.Text, password.Text);
        }

        void m_ftp_Connected(FTP source)
        {
            // when this happens we're ready to send command
            OnResponse("Connected.");
        }

        void m_ftp_ResponseReceived(FTP source, FTPResponse Response)
        {
            OnResponse(Response.Text);
        }

        private void OnResponse(string response)
        {
            if (this.InvokeRequired)
            {
                this.Invoke(new StringDelegate(OnResponse), new object[] { response } );
                return;
            }
            ListViewItem item = new ListViewItem(new string[] { DateTime.Now.ToShortTimeString(), response });
            status.Items.Insert(0, item);
            status.Columns[1].Width = -1;
        }

        private void getFileList_Click(object sender, EventArgs e)
        {
            FTPFiles files = m_ftp.EnumFiles();

            fileList.Items.Clear();

            foreach (FTPFile file in files)
            {
                fileList.Items.Add( new ListViewItem( new string[] { file.Name, file.Size.ToString() } ));
            }

            tabs.SelectedIndex = 1;
        }

        private void upload_Click(object sender, EventArgs e)
        {
            FileStream stream = File.OpenRead("\\My Documents\\My Pictures\\Waterfall.jpg");
            m_ftp.SendFile(stream, "waterfall.jpg");
            stream.Close();
        }
    }
}